### PR TITLE
ENH: Add support for nullable boolean and integers in Stata writers

### LINKED ISSUE #40855: ENH: Add support for nullable integer columns to StataWriter
When attempting to export a dataframe to a Stata data file, I hit the following error due to my dataframe including variables of the Int64 dtype. 

```python
import pandas as pd

# CREATE DATAFRAME
s1 = pd.Series(2**33, dtype='Int64')
s2 = pd.Series("ABC", dtype=object)
df = pd.DataFrame({'col1': s1, 'col2': s2})

df = df.append({'col1': None, 'col2': "DEF"}, ignore_index=True)

# EXPORT TO STATA
df.to_stata('df.dta', version=117)
```
Error message: 
> NotImplementedError: Data type Int64 not supported.

My expectation is columns of dtype Int64 (i.e., a nullible INT) would end up as byte, int or float in the dta file depending on values therein.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 67a3d4241ab84419856b84fc3ebc9abcbe66c6b3
python           : 3.8.6.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.14393
machine          : AMD64
processor        : Intel64 Family 6 Model 61 Stepping 2, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : en
LOCALE           : English_United States.1252

pandas           : 1.1.4
numpy            : 1.19.4
pytz             : 2020.4
dateutil         : 2.8.1
pip              : 20.2.4
setuptools       : 50.3.1.post20201107
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : 3.4.3
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.19.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.3
numexpr          : None
odfpy            : None
openpyxl         : 3.0.5
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : 1.5.3
sqlalchemy       : 1.3.20
tables           : None
tabulate         : None
xarray           : None
xlrd             : 2.0.1
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by bashtage**
Add code that allows nullable arrays to be written

closes #40855

- [X] closes #40855
- [X] tests added / passed
- [X] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [X] whatsnew entry

### CODE REVIEW

**Comment by bashtage** [comment]
Seeing a lot of random failures unrelated to this.

**Comment by bashtage** [comment]
@jreback I think this is ready, and all green now.

File: doc/source/whatsnew/v1.4.0.rst [hunk 1]
```diff
@@ -101,14 +101,18 @@ Other enhancements
 - :class:`DataFrameGroupBy` operations with ``as_index=False`` now correctly retain ``ExtensionDtype`` dtypes for columns being grouped on (:issue:`41373`)
 - Add support for assigning values to ``by`` argument in :meth:`DataFrame.plot.hist` and :meth:`DataFrame.plot.box` (:issue:`15079`)
 - :meth:`Series.sample`, :meth:`DataFrame.sample`, and :meth:`.GroupBy.sample` now accept a ``np.random.Generator`` as input to ``random_state``. A generator will be more performant, especially with ``replace=False`` (:issue:`38100`)
+- Additional options added to :meth:`.Styler.bar` to control alignment and display, with keyword only arguments (:issue:`26070`, :issue:`36419`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.4.0.rst:104]: extra?

File: doc/source/whatsnew/v1.4.0.rst [hunk 2]
```diff
@@ -101,14 +101,18 @@ Other enhancements
...
 - :meth:`.GroupBy.cummin` and :meth:`.GroupBy.cummax` now support the argument ``skipna`` (:issue:`34047`)
 - :meth:`read_table` now supports the argument ``storage_options`` (:issue:`39167`)
 - :meth:`DataFrame.to_stata` and :meth:`StataWriter` now accept the keyword only argument ``value_labels`` to save labels for non-categorical columns
 - Methods that relied on hashmap based algos such as :meth:`DataFrameGroupBy.value_counts`, :meth:`DataFrameGroupBy.count` and :func:`factorize` ignored imaginary component for complex numbers (:issue:`17927`)
 - Add :meth:`Series.str.removeprefix` and :meth:`Series.str.removesuffix` introduced in Python 3.9 to remove pre-/suffixes from string-type :class:`Series` (:issue:`36944`)
 - Attempting to write into a file in missing parent directory with :meth:`DataFrame.to_csv`, :meth:`DataFrame.to_html`, :meth:`DataFrame.to_excel`, :meth:`DataFrame.to_feather`, :meth:`DataFrame.to_parquet`, :meth:`DataFrame.to_stata`, :meth:`DataFrame.to_json`, :meth:`DataFrame.to_pickle`, and :meth:`DataFrame.to_xml` now explicitly mentions missing parent directory, the same is true for :class:`Series` counterparts (:issue:`24306`)
-
+-
```
Comment (jreback) [inline doc/source/whatsnew/v1.4.0.rst:113]: extra line here

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm, doc typo. ping on green.

**Comment by jreback** [comment]
pls merge master as well

File: doc/source/whatsnew/v1.4.0.rst (hunk 1 above)
Comment (bashtage) [reply doc/source/whatsnew/v1.4.0.rst]: Must have been a rebase error. Thanks.

**Comment by jreback** [comment]
thanks @bashtage

**Pull request merged**